runtime/lua/vim/_editor.lua

```diff
@@ -272,6 +272,8 @@ vim.fn = setmetatable({}, {
       _fn = function()
         error(string.format('Tried to call API function with vim.fn: use vim.api.%s instead', key))
       end
+    elseif vim.call("exists", "*" .. key) == 0 then
```

Comment:
should check `t[key]` first, no need to check this if we already stored a value there.

```suggestion
    elseif t[key] == nil and vim.call("exists", "*" .. key) == 0 then
```

Reply:
I think `t[key]` would call this `__index` metamethod and end up in a infinite recursive loop. `__index` is only called when the key does not exist on a table.